src/app/api/auth/route.ts

```diff
@@ -2,15 +2,26 @@
 import { NextResponse } from "next/server";
 import { cookies } from 'next/headers'
 
+//function to generate a random string with a given length
+function generateRandomString(length: number) {
```

Comment:
One of the return types inside apiFunctions is Promise<Map<string, number> | undefined>. is this proper or do I have to define it so that it only returns Promise<Map<string, number>>?